### DISCUSSION TITLE
Building a library with both x86-64 and arm64 slices

### DISCUSSION THREAD

**Discussion by mattmassicotte** [Homebrew/discussions#16] [Everyday usage] [answered]
Hello!

I've got a few dependencies that I need to build fat, with both x86-64 and arm64 slices. After some googling, it looks like this might have been possible in the past, but could no longer be supported.

Right now, I can do what I need by manually tweaking formula. Is there any way I can customize the install process to do this in a more automated way?

**Comment by SMillerDev** [comment, marked as answer]
No, homebrew doesn't support fat binaries and probably never will. Like you mentioned it used to be supported but it was way more trouble then it was worth. For the upcoming arm64 world we plan on having separate binaries.

**Reply by mattmassicotte** [reply]
Ok, thank you! That's great to know. I was hunting around for information about how arm64 might be handled. Do you have any links to issues/documentation that I could read up on how multiple binaries might work?

**Reply by SMillerDev** [reply]
Not really, I think it was mentioned briefly in the ARM testing issue in homebrew/brew. But the way it would work is the same as bottles work right now, it would just be `"SOME_HASH" => :big_sur_arm64` AFAIK. You'd have to dig through some code to figure it out.

**Comment by Irev-Dev** [comment]
Hey, I'm looking to create a Tap, first time doing anything homebrew related. Wanted to clarify, is there a way to supply multiple binaries?
and if not should I default to intel / x86?

**Reply by SMillerDev** [reply]
You can provide binaries for multiple platforms, you just can't have Homebrew cross-compile
